lib/ansible/modules/cloud/smartos/imgadm.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Jasper Lievisse Adriaanse <j@jasper.la>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: imgadm
+short_description: Manage SmartOS images
+description:
+    - Manage SmartOS virtual machine images through imgadm(1M)
+version_added: "2.3"
+author: Jasper Lievisse Adriaanse (@jasperla)
+options:
+    force:
+        required: false
+        choices: [ yes, no ]
+        description:
+        - Force a given operation (where supported by imgadm(1M)).
+    pool:
+        required: false
+        default: zones
+        description:
+        - zpool to import to or delete images from.
+    source:
+        required: false
+        description:
+        - URI for the image source.
+    state:
+        required: true
+        choices: [ present, absent, deleted, imported, updated, vacumed ]
+        description:
+        - State the object operated on should be in. C(imported) is an alias for
+          for C(present) and C(deleted) for C(absent). When set to C(vacumed)
+          and C(uuid) to C(*), it will remove all unused images.
+    type:
+        required: false
+        choices: [ imgapi, docker, dsapi ]
+        default: imgapi
+        description:
+        - Type for image sources.
+    uuid:
+        required: false
+        description:
+        - Image UUID. Can either be a full UUID or C(*) for all images.
```

Comment:
If this requires a higher python version than 2.4 then there needs to be a requirements section that lists it (for instance):
```
    requirements:
        - python >= 2.6
```